packages/langium-cli/src/generator/type-collector.ts

```diff
@@ -153,51 +269,38 @@ function collectAllParserRules(documents: LangiumDocuments, grammars: langium.Gr
         }
         visited.add(doc.uri);
         for (const rule of grammar.rules) {
-            if (langium.isParserRule(rule) && !rule.fragment && !isDataTypeRule(rule)) {
-                rules.add(rule);
+            if (langium.isParserRule(rule) && !rule.fragment) {
+                isDataTypeRule(rule) ? astResources.datatypeRules.add(rule) : astResources.parserRules.add(rule);
```

Comment:
Minor: Please don't use ternary operations to do control flow. Use `if ... else` instead.